### PR TITLE
devfile delete Integration test on kubernetes cluster

### PR DESCRIPTION
**Pull request by prietyc123**
**What type of PR is this?**

/kind test

**What does does this PR do / why we need it**:

Runs the devfile delete integration test on kubernetes cluster.

**Which issue(s) this PR fixes**:

Fixes - Part of #2752

**How to test changes / Special notes to the reviewer**:

The test should pass on travis with ```make test-cmd-devfile-delete``` command.

### CODE REVIEW

**Comment by prietyc123** [comment]
Blocked with devfile deletion test on kuberenetes which has been jotted down in descriptive manner in issue : https://github.com/openshift/odo/issues/2928

```
$ git clone https://github.com/che-samples/web-nodejs-sample.git context/projectDir
Cloning into '/tmp/394302036/projectDir'...
Setting current dir to: /tmp/394302036/projectDir

$ odo create nodejs --project namespace componentName
Experimental mode is enabled, use at your own risk

 Validation
 •  Checking devfile compatibility  ...
 ✓  Checking devfile compatibility [33281ns]
[odo]  •  Validating devfile component  ...
 ✓  Validating devfile component [64477ns]
[odo] 
[odo] Please use `odo push` command to create the component with source deployed

$ odo push --devfile devfile.yaml --namespace namespace
 •  Push devfile component umffvx  ...
✗  Failed to start component with name umffvx.
Error: Failed to create the component: unable to create or update component: unable to create Deployment umffvx: the server could not find the requested resource
Deleting project: zxobygmpyf
```

**Comment by prietyc123** [comment]
> Blocked with devfile deletion test on kuberenetes which has been jotted down in descriptive manner in issue : #2928
> 
> ```
> $ git clone https://github.com/che-samples/web-nodejs-sample.git context/projectDir
> Cloning into '/tmp/394302036/projectDir'...
> Setting current dir to: /tmp/394302036/projectDir
> 
> $ odo create nodejs --project namespace componentName
> Experimental mode is enabled, use at your own risk
> 
>  Validation
>  •  Checking devfile compatibility  ...
>  ✓  Checking devfile compatibility [33281ns]
> [odo]  •  Validating devfile component  ...
>  ✓  Validating devfile component [64477ns]
> [odo] 
> [odo] Please use `odo push` command to create the component with source deployed
> 
> $ odo push --devfile devfile.yaml --namespace namespace
>  •  Push devfile component umffvx  ...
> ✗  Failed to start component with name umffvx.
> Error: Failed to create the component: unable to create or update component: unable to create Deployment umffvx: the server could not find the requested resource
> Deleting project: zxobygmpyf
> ```

Resolved with adding latest minikube on travis CI.

**Comment by prietyc123** [comment]
prow failure 

/retest

**Comment by prietyc123** [comment]
Flake on [travis CI ](https://travis-ci.com/github/openshift/odo/jobs/328519613)tracked in [issue](https://github.com/openshift/odo/issues/3099) 

Restarted the job.

**Comment by prietyc123** [comment]
Failure from tracked issue. See comment https://github.com/openshift/odo/issues/2877#issuecomment-624514156

/retest

**Comment by amitkrout** [comment]
I was following the changes since you started working on it. Looks good to me 👍
/lgtm

**Comment by amitkrout** [comment]
Rebased
/lgtm

File: tests/integration/devfile/cmd_devfile_delete_test.go [hunk 1]
```diff
@@ -14,32 +14,47 @@ import (
...
-		helper.Chdir(context)
-
 		os.Setenv("GLOBALODOCONFIG", filepath.Join(context, "config.yaml"))
 
 		// Devfile commands require experimental mode to be set
 		helper.CmdShouldPass("odo", "preference", "set", "Experimental", "true")
+
+		if os.Getenv("KUBERNETES") == "true" {
+			homeDir := helper.GetUserHomeDir()
+			kubeConfigFile := helper.CopyKubeConfigFile(filepath.Join(homeDir, ".kube", "config"), filepath.Join(context, "config"))
```
Comment (kadel) [inline tests/integration/devfile/cmd_devfile_delete_test.go:38]: discussion in https://github.com/openshift/odo/pull/3041#pullrequestreview-406508839

**Comment by girishramnani** [comment]
Same comments as https://github.com/openshift/odo/pull/3041

**Comment by prietyc123** [comment]
prow failure 

/retest

**Comment by prietyc123** [comment]
> Same comments as #3041

Done in #3041 and this pr is rebased with the changes.

**Comment by prietyc123** [comment]
Failure https://deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gcs/origin-ci-test/pr-logs/pull/openshift_odo/2913/pull-ci-openshift-odo-master-v4.2-integration-e2e/928#1:build-log.txt%3A592 

issue tracked in https://github.com/openshift/odo/issues/3256

/retest

**Pull request merged**